src/tests/ftest/util/command_utils.py

```diff
@@ -183,32 +215,63 @@ def stop(self):
 
         """
         if self._process is not None:
-            # Use a list to send signals to the process with one second delays:
-            #   Interupt + wait 3 seconds
-            #   Terminate + wait 2 seconds
-            #   Quit + wait 1 second
-            #   Kill
-            signal_list = [
-                signal.SIGINT, None, None, None,
-                signal.SIGTERM, None, None,
-                signal.SIGQUIT, None,
-                signal.SIGKILL]
+            # Send a SIGTERM to the stop the subprocess and if it is still
+            # running after 5 seconds send a SIGKILL and report an error
+            signal_list = [signal.SIGTERM, signal.SIGKILL]
 
             # Turn off verbosity to keep the logs clean as the server stops
             self._process.verbose = False
 
-            # Keep sending signals and or waiting while the process is alive
+            # Send signals while the process is still running
+            state = None
             while self._process.poll() is None and signal_list:
-                signal_type = signal_list.pop(0)
-                if signal_type is not None:
-                    self._process.send_signal(signal_type)
+                signal_to_send = signal_list.pop(0)
+                msg = "before sending signal {}".format(signal_to_send)
+                state = self.get_subprocess_state(msg)
+                self.log.info(
+                    "Sending signal %s to %s (state=%s)", str(signal_to_send),
+                    self._command, str(state))
+                self._process.send_signal(signal_to_send)
                 if signal_list:
-                    time.sleep(1)
+                    time.sleep(5)
+
             if not signal_list:
-                # Indicate an error if the process required a SIGKILL
-                raise CommandFailure("Error stopping '{}'".format(self))
+                if state and len(state) == 1 and state[0] not in ("D", "Z"):
+                    # Indicate an error if the process required a SIGKILL and
+                    # either multiple processes were still found running or the
+                    # parent process was in any state accept uninterruptible
+                    # sleep (D) or zombie (Z).
+                    raise CommandFailure("Error stopping '{}'".format(self))
```

Comment:
I don't understand this clause... It seems like we'd want to fail the test regardless of the lingering process state, right? If one of our processes is stuck in D or Z it seems likely due to a bug. Also: s/accept/except/

Reply:
What we found in the previous [runs](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2522/11/artifact/Functional/job-2020-05-01T01.38-3aeea5b-container-simple_create_delete_test/job.log) was the daos_agent or daos[_io]_server processes were correctly being terminated when the `SIGTERM` signal was sent to the orterun process.   The issue was the orterun process was still running in either the `D` or `Z` state.  So if this situation occurs, we don't want want to report it as an error.  The `state` list contains the state of every local process (orterun) and any a single `R` for any remote processes found.  So if this list equals `['D']` or `['Z']` then we don't want to mark the test as `ERROR` even though we sent the `SIGKILL` signal.

Thanks for catching the comment typo.  I'll fix it if I need push any more changes.

Reply:
[Here's](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2522/11/artifact/Functional/job-2020-05-01T01.47-1db8df3-object-array_obj_test/job.log) an example of the server stop failure with extra process info.

Reply:
Ah, OK. That makes sense. Interesting that orterun hangs like that. Nice to see that the code for capturing process info in cleanup works, too.